src/main/scala/com/newlibertie/pollster/impl/Poll.scala

```diff
@@ -28,16 +31,49 @@ object Poll {
     }
     new Poll(pollParameters, cryptographicParameters)
   }
-
-  def write(poll: Poll) = {
-    DataAdapter.createPoll(poll)
-  }
-
   def read(id:String) = {
-    DataAdapter.getPoll(id)
-
+    DataAdapter.getPoll(id) match {
+      case m:mutable.Map[String, Any] =>
+        new Poll(PollParameters(Some(m.get("id").toString),
+          m.get("title").toString,
+          parse(m.get("tags").get.toString).values.asInstanceOf[List[String]],
+          m.get("creator_id").toString,
+          m.get("opening_ts").get.asInstanceOf[Date],
+          m.get("closing_ts").get.asInstanceOf[Date],
+          Some(m.get("creation_ts").get.asInstanceOf[Date]),
+          Some(m.get("last_modification_ts").get.asInstanceOf[Date]),
+          m.get("poll_spec").get.toString,
+          m.get("poll_type").get.toString),
+        new CryptographicParameters(new BigInteger(m.get("large_prime_p").get.toString),
+          new BigInteger(m.get("generator_g").get.toString),
+          new BigInteger(m.get("private_key_s").get.toString)))
+      case _ =>  -1
+    }
   }
 }
 
 class Poll(val p:PollParameters, val cp:CryptographicParameters = new CryptographicParameters()) {
+  def create() = {
+    DataAdapter.createPoll(this)
+  }
+  def update() = {
+    DataAdapter.updatePoll(this)
+  }
+  def canDelete():Boolean = {
+    p.opening_ts before new Date() // TODO check if any ballot has been cast to disallow
+  }
+  def canClose():Boolean = {
+    val cur_ts = new Date()
+    (p.opening_ts before cur_ts) && (p.closing_ts after cur_ts)
+  }
+  def canOpen():Boolean = {
+    p.opening_ts after new Date()
+  }
+  def deletePoll(): Int = {
+    DataAdapter.deletePoll(p.id.get)
+  }
+  override def toString: String = {
```

Comment:
looks like you are buidling a specific kind of serialization.  if that is so can we name it more specifically please

Reply:
How about using toJsonString? Else can you suggest a name as it is used to return Poll object to the REST client?